crates/relayer/src/config.rs

```diff
@@ -50,6 +53,35 @@ impl Display for GasPrice {
     }
 }
 
+impl TryFrom<String> for GasPrice {
+    type Error = ConfigError;
+
+    fn try_from(price: String) -> Result<Self, Self::Error> {
```

Comment:
This seems like a good first implementation of parsing from String -> GasPrice!

For reference, had a quick look through the SDK and could not find a good example that can help guide us in the future. Closes thing I could find was parsing from strings into decimal coins (eg `10uatoms`). This doesn't work for what we need, which is parsing into f64 coins (`0.03uatoms`).

https://github.com/cosmos/cosmos-sdk/blob/main/types/dec_coin.go#L624-L631

---
That being said, I wasn't very comfortable with this code without a sanity check on it, esp. given the note on "Note: `split_once` does _not_ split inclusively". Wrote a simple test:


```rust
    #[test]
    fn gas_price_try_from() {
        let gp_original = GasPrice::new(10.0, "atom".to_owned());

        let gp_raw: String = gp_original.to_string();
        let gp: GasPrice = gp_raw.try_into().expect("could not parse String into GasPrice");

        assert_eq!(gp, gp_original);
    }
```
seems to fail

```
% cargo test --package ibc-relayer --lib -- config::tests::gas_price_try_from --exact --nocapture
   Compiling ibc-relayer v0.20.0 (/Users/adiseredinschi/Hammers/hermes/crates/relayer)
    Finished test [unoptimized + debuginfo] target(s) in 23.06s
     Running unittests src/lib.rs (target/debug/deps/ibc_relayer-bc3285dcfe99ee96)

running 1 test
thread 'config::tests::gas_price_try_from' panicked at 'assertion failed: `(left == right)`
  left: `GasPrice { price: 10.0, denom: "tom" }`,
 right: `GasPrice { price: 10.0, denom: "atom" }`', crates/relayer/src/config.rs:553:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test config::tests::gas_price_try_from ... FAILED
```


See https://github.com/informalsystems/hermes/commit/3915ad2868fe030c7cc443bf4b7642ef9d821934.

Reply:
Update: we can use a different solution: b9966a02